### PR TITLE
Add advisory and troubleshooting on non-tls mode for machineid kube

### PR DESCRIPTION
**Pull request by stevenGravy**
PUtting in until #19811 is closed.

In the GitHub Actions K8s and K8s guide for machineid prereqs.

image

### CODE REVIEW

File: docs/pages/machine-id/troubleshooting.mdx
Comment (strideynet) [inline docs/pages/machine-id/troubleshooting.mdx:296]:
```suggestion
## Kubernetes connections are failing with `Unable to connect to the server: x509: certificate signed by unknown authority`
```

File: docs/pages/includes/machine-id/kubernetes-machineidnote.mdx
Comment (alexfornuto) [inline docs/pages/includes/machine-id/kubernetes-machineidnote.mdx:9]:
```suggestion
command with your proxy address:
```

File: docs/pages/includes/machine-id/kubernetes-machineidnote.mdx
Comment (alexfornuto) [inline docs/pages/includes/machine-id/kubernetes-machineidnote.mdx:15]: Can you expand on "this will apply" to specifically state if TLS routing is enabled or not?

File: docs/pages/machine-id/troubleshooting.mdx
Comment (alexfornuto) [inline docs/pages/machine-id/troubleshooting.mdx:300]: Since 'cluster' is overloaded, should we be specific?

```suggestion
A self-hosted Teleport cluster is connecting Machine ID to Kubernetes clusters with the following errors.  This can happen for non-TLS configured 
```

File: docs/pages/machine-id/troubleshooting.mdx
Comment (alexfornuto) [inline docs/pages/machine-id/troubleshooting.mdx:301]: Line breaks seems to be random throughout. Should be standardized at ~80 chars. This is one example, but please adjust throughout.

```suggestion
A self-hosted cluster is connecting Machine ID to Kubernetes clusters with the
following errors.  This can happen for non-TLS configured Teleport clusters.
```

File: docs/pages/machine-id/troubleshooting.mdx
Comment (alexfornuto) [inline docs/pages/machine-id/troubleshooting.mdx:310]: 1. check the `proxy.tls_routing_enabled` _what_? Key? Check the value of that key/config?
2. Move the explanation of the output to after the command, and use a `:` before.
3. Keep newlines around code blocks. If/when (most likely when) we move to tooling with standard MDX rendering, adherence to Markdown standards will matter.

````suggestion
To confirm the TLS routing mode check the value of the `proxy.tls_routing_enabled` key with this command, substituting your proxy address:

```bash
curl https://teleport.example.com:443/webapi/ping | jq
```

If the value is `false` then this is a non-TLS routing configuration.
````

File: docs/pages/machine-id/troubleshooting.mdx
Comment (alexfornuto) [inline docs/pages/machine-id/troubleshooting.mdx:315]:
```suggestion
Proxies configured with non-TLS routing"  use specific ports for various types
of traffic. That requires that a Kubernetes
```

Requires additional line break adjustments for the rest of the paragraph.

File: docs/pages/machine-id/troubleshooting.mdx
Comment (alexfornuto) [inline docs/pages/machine-id/troubleshooting.mdx:316]:
```suggestion
connection use its designated port. Currently Machine ID requires that the Kubernetes public address is set
```

or 

```suggestion
connection use its designated port. Currently Machine ID requires the Kubernetes public address to be set
```

File: docs/pages/machine-id/troubleshooting.mdx
Comment (alexfornuto) [inline docs/pages/machine-id/troubleshooting.mdx:322]: Within the `proxy_service` _what_, key? Also, `s/proxy/Teleport Proxy service/`.

File: docs/pages/machine-id/troubleshooting.mdx
Comment (alexfornuto) [inline docs/pages/machine-id/troubleshooting.mdx:332]: Again, if using "the" name the type of thing you're specifying with `kube_public address`.

**Review by alexfornuto — CHANGES_REQUESTED** [review]
In addition to line-specific notes, there's use of `jq` throughout on the assumption that the user has it installed, but it's not provided by default in macOS or common Linux distros. I have no idea about Windows.

File: docs/pages/machine-id/troubleshooting.mdx
Comment (stevenGravy) [reply docs/pages/machine-id/troubleshooting.mdx]: Please see update

File: docs/pages/machine-id/troubleshooting.mdx
Comment (stevenGravy) [reply docs/pages/machine-id/troubleshooting.mdx]: Please see update

File: docs/pages/machine-id/troubleshooting.mdx
Comment (stevenGravy) [reply docs/pages/machine-id/troubleshooting.mdx]: Please see the udpates

**Comment by stevenGravy** [comment]
The comments should be addressed @alexfornuto

**Comment by zmb3** [comment]
Can we squash these commits and clean up the commit message a bit too?

**Comment by stevenGravy** [comment]
> In addition to line-specific notes, there's use of `jq` throughout on the assumption that the user has it installed, but it's not provided by default in macOS or common Linux distros. I have no idea about Windows.

I have mentioned it as a optional too.

**Comment by stevenGravy** [comment]
> Can we squash these commits and clean up the commit message a bit too?

I don't see any options on that in the UI.  Is there something we can do from CLI on that.

**Comment by zmb3** [comment]
Yes, you'll need to do an interactive rebase.

File: docs/pages/includes/machine-id/kubernetes-machineidnote.mdx
Comment (stevenGravy) [reply docs/pages/includes/machine-id/kubernetes-machineidnote.mdx]: Please see updates

File: docs/pages/includes/machine-id/kubernetes-machineidnote.mdx
Comment (alexfornuto) [inline docs/pages/includes/machine-id/kubernetes-machineidnote.mdx:13]:
````suggestion
```code
$ curl https://teleport.example.com:443/webapi/ping | jq
```
````

File: docs/pages/includes/machine-id/kubernetes-machineidnote.mdx
Comment (alexfornuto) [inline docs/pages/includes/machine-id/kubernetes-machineidnote.mdx:15]:
```suggestion
The optional tool [`jq`](https://stedolan.github.io/jq/) is used here to help display the JSON output.
```

File: docs/pages/includes/machine-id/kubernetes-machineidnote.mdx
Comment (alexfornuto) [inline docs/pages/includes/machine-id/kubernetes-machineidnote.mdx:21]:
```suggestion
The `kube_public_addr` is set within the `proxy_service` by Teleport administrators:
```

File: docs/pages/machine-id/troubleshooting.mdx
Comment (alexfornuto) [inline docs/pages/machine-id/troubleshooting.mdx:301]:
```suggestion
with the following errors. This can happen for non-TLS configured Teleport clusters.
```

File: docs/pages/machine-id/troubleshooting.mdx
Comment (alexfornuto) [inline docs/pages/machine-id/troubleshooting.mdx:319]:
```suggestion
Proxies configured with non-TLS routing use specific ports for various types
```

File: docs/pages/machine-id/troubleshooting.mdx
Comment (alexfornuto) [inline docs/pages/machine-id/troubleshooting.mdx:315]:
```suggestion
```

File: docs/pages/machine-id/troubleshooting.mdx
Comment (alexfornuto) [reply docs/pages/machine-id/troubleshooting.mdx]: The update does not address these concerns.

File: docs/pages/machine-id/troubleshooting.mdx
Comment (alexfornuto) [reply docs/pages/machine-id/troubleshooting.mdx]: Update does not resolve.

**Comment by alexfornuto** [comment]
@zmb3 Regarding the commit messages, don't we force squash commits anyway?

File: docs/pages/machine-id/troubleshooting.mdx
Comment (alexfornuto) [reply docs/pages/machine-id/troubleshooting.mdx]: The update does not address these concerns.

**Comment by zmb3** [comment]
@alexfornuto the merge queue will automatically generate a commit message based on the 18 commits in this PR. That means the commit message will be littered with things like `update note`, `spacing`, and `fix header`.

That's not a high quality commit message.

File: docs/pages/machine-id/troubleshooting.mdx
Comment (stevenGravy) [reply docs/pages/machine-id/troubleshooting.mdx]: Please see latest

File: docs/pages/machine-id/troubleshooting.mdx
Comment (stevenGravy) [reply docs/pages/machine-id/troubleshooting.mdx]: Please see latest

**Comment by stevenGravy** [comment]
> @alexfornuto the merge queue will automatically generate a commit message based on the 18 commits in this PR. That means the commit message will be littered with things like `update note`, `spacing`, and `fix header`.
> 
> That's not a high quality commit message. Rather than a laundry list of random changes we made along the way, we should rebase the commits and have a single descriptive message describing the change and the reasoning behind it.

Once we get approval I can submit another PR then.

**Review by alexfornuto — APPROVED** [review]
Approved and rebased.

**Review by strideynet — APPROVED** [review]

**Comment by stevenGravy** [comment]
> Approved and rebased.

Huge help @alexfornuto !!

**Pull request merged**
